pkg/reconciler/taskrun/resources/cloudevent/cloud_event_controller.go

```diff
@@ -67,6 +69,15 @@ func cloudEventDeliveryFromTargets(targets []string) []v1alpha1.CloudEventDelive
 // the TaskRun is complete. `tr` is used to obtain the list of targets but also
 // to construct the body of the
 func SendCloudEvents(tr *v1alpha1.TaskRun, ceclient CEClient, logger *zap.SugaredLogger) error {
+	logger = logger.With(zap.String("taskrun", tr.Name))
+
+	// Make the event we would like to send:
+	event, err := EventForTaskRun(tr)
```

Comment:
Nice, you separated creating the event from the delivery. Is this so we only build it once, in case of multiple deliveries?

Reply:
yeah, the controller now uses the client directly, and all the custom stuff is doing is understanding how to take a `TaskRun` and turn it into an `event.Event`.